libs/reconciler/mocks_test.go

```diff
@@ -87,15 +87,15 @@ func (m *mockConfigurator) Create(ctx context.Context, item depgraph.Item) (err
 	}
 	if mItem.asyncCreate {
 		done := reconciler.ContinueInBackground(ctx)
-		go func() {
+		go func(err error) {
 			select {
 			case <-time.After(asyncOpDuration):
 				break
 			case <-ctx.Done():
 				err = errors.New("failed to complete")
 			}
 			done(err)
-		}()
+		}(err)
```

Comment:
Is the intent here to have err be a local variable? You are not using it as an input argument.

If a local variable it would be more clear to have "var err error" at the top of the func.

Reply:
It is used as the input argument when `<-time.After(asyncOpDuration)` fires.
Some background: this is a mock simulating async operation. Used in unit test where I specify what error I want to have returned and then I check that reconciler handled it correctly.

Reply:
Sorry I couldn't read correctly, maybe due to not understanding the purpose of the code.

Maybe some comment (or renaming) would be helpful, because A Done channel typically doesn't imply a failure.
Calling the channel FaultInjection and having the error be "Received injected fault" seems to match the explanation you wrote above. Alternatively a Cancel channel if there are cases when you force an operation to terminate early.

Reply:
No, while the word `Done()` has "positives vibes", it is often used to limit the time allocated for a work. If you look [at the docs ](https://pkg.go.dev/context#Context) it says:
```
Done returns a channel that's closed when work done on behalf of this context should be canceled ...
```
[Here](https://www.sohamkamani.com/golang/context-cancellation-and-values/#listening-for-cancellation) is some example from the web.

Reply:
But I can rename `err` to `injectedErr` to make the purpose of this code more clear.

Reply:
Such a rename would help.

Reply:
Done